File: 2020/2020W50/PreppindataW50.py
import pandas as pd
import re

xlsx = pd.ExcelFile("F:/Data/Secret Santa.xlsx")

final = pd.read_excel(xlsx,sheet_name=0)
final['Email'] = final['Email'].apply(lambda x: re.sub('(.*)@([a-z]+).([a-z]+)','\\1@\\2.\\3',x))
final['Secret Santee'] = (final.sort_values(by=['Secret Santa'], ascending=True)['Secret Santa'].shift(-1, fill_value=min(final['Secret Santa'])))
final['Email Subject'] = 'Secret Santa🤫🎅'
final['Email Body'] = final['Secret Santa']+', the results are in, your secret santee is: '+final['Secret Santee']+'. Good luck finding a great gift!'
final = final[['Email', 'Email Subject', 'Email Body']]

File: 2020/2020W50/preppindataw50.R
library(readxl)
library(dplyr)
library(stringr)

final <- read_excel( "F:/Data/Secret Santa.xlsx") %>%
  arrange(.,`Secret Santa`) %>%
  mutate('Secret Santee' = lead(`Secret Santa`,default = min(`Secret Santa`)),
         'Email' = str_replace_all(`Email`, '(.*)@([a-z]+).([a-z]+)','\\1@\\2.\\3'),
         'Email Subject' = 'Secret Santa????????',
         'Email Body' = paste0(`Secret Santa`,', the results are in, your secret santee is: ',`Secret Santee`,'. Good luck finding a great gift!')) %>%
  select(c('Email', 'Email Subject', 'Email Body'))

View(final)
